pkg/storage/storageclass.go

```diff
@@ -0,0 +1,266 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "k8s.io/api/core/v1"
+	storageV1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StorageClassBuilder provides struct for storageclass object containing
+// connection to the cluster and the storageclass definitions.
+type StorageClassBuilder struct {
+	// Storageclass definition. Used to create the storageclass object.
+	Definition *storageV1.StorageClass
+	// Created storageclass object
+	Object *storageV1.StorageClass
+	// Used in functions that define or mutate storageclass definition. errorMsg is processed before the storageclass
+	// object is created.
+	errorMsg  string
+	apiClient *clients.Settings
+}
+
+// AdditionalOptions additional options for storageclass object.
+type AdditionalOptions func(builder *StorageClassBuilder) (*StorageClassBuilder, error)
+
+// NewStorageClassBuilder creates a new instance of StorageClassBuilder.
+func NewStorageClassBuilder(apiClient *clients.Settings, name, provisioner string) *StorageClassBuilder {
```

Comment:
I think it's name duplication. You can simply re-name it to NewClassBuilder because when you call this function you have storage in package name.
`storageClass := storage.NewClassBuilder`